checkov/sast/engines/prisma_engine.py

```diff
@@ -351,8 +356,83 @@ def create_report(self, prisma_report: PrismaReport) -> List[SastReport]:
+                sast_rule_matches: Dict[str, RuleMatch] = {}
+                for policy_id, rule_match in rule_matches.items():
+                    if rule_match.metadata.framework != 'CDK':
+                        sast_rule_matches[policy_id] = rule_match
+                        continue
+                    if lang not in [c.language for c in cdk_reports]:
+                        new_cdk_report = PrismaReport(rule_match={lang: {}}, errors=report.sast_report.errors,
+                                                  profiler=report.sast_report.profiler,
+                                                  run_metadata=report.sast_report.run_metadata,
+                                                  imports={}, reachability_report={})
+                        new_report = CDKReport(f'cdk_{lang.value}', report.sast_report.run_metadata, lang, new_cdk_report)
+                        cdk_reports.append(new_report)
+                    for cdk_report in cdk_reports:
+                        if cdk_report.language == lang:
+                            cdk_report.cdk_report.rule_match[lang][policy_id] = rule_match
+                            for failed_check in report.failed_checks:
+                                if failed_check.check_id == policy_id:
+                                    cdk_report.failed_checks.append(failed_check)
+                                
+                            for skiped_check in report.skipped_checks:
+                                if skiped_check.check_id == policy_id:
+                                    cdk_report.skipped_checks.append(skiped_check)
+                            break
+                if sast_rule_matches:
+                    report.sast_report.rule_match[lang] = sast_rule_matches
+                else:
+                    report.sast_report.rule_match = {}
+                self._update_sast_report_checks(report, cdk_reports)
+
+        return self._get_all_reports()
+
+    def _update_sast_report_checks(self, report: SastReport, cdk_reports: List[CDKReport]) -> None:
+        sast_failed_checks = []
+        sast_skiped_checks = []
+
+        for fail_check in report.failed_checks:
+            for cdk_report in cdk_reports:
+                if report.language == cdk_report.language and fail_check.check_id not in [f.check_id for f in cdk_report.failed_checks]:
+                    sast_failed_checks.append(fail_check)
+                    break
+            else:
+                sast_failed_checks = report.failed_checks
+                break
+
+        for skip_check in report.skipped_checks:
+            for cdk_report in cdk_reports:
+                if report.language == cdk_report.language and skip_check.check_id not in [s.check_id for s in cdk_report.skipped_checks]:
+                    sast_skiped_checks.append(skip_check)
+                    break
+            else:
+                sast_skiped_checks = report.skipped_checks
+                break
```

Comment:
the failed/skipped logic looks the same, put it into a function that iterates on a check lists